src/gluonts/distribution/binned.py

```diff
@@ -174,6 +174,10 @@ def s(bin_probs):
 
         return _sample_multiple(s, self.bin_probs, num_samples=num_samples)
 
+    @property
+    def args(self) -> List:
```

Comment:
The property is defined in the base class already.

> You impose a constraint on the arguments of all distributions that they are all sliceable.

I agree, but that's more about the `slice_axis` method (we should be careful there I guess) than `args`.

> Also, by using a list order is important and can easily mixed up.

Do you suggest using a dictionary?

Reply:
I guess it's fine for now, although I have some concerns about this axis thing.

Reply:
A less ambiguous solution (but a more verbose one) would be to define the `slice_axis` method for each distribution. A somewhat compromise could be to have this as fallback definition, and specialize `slice_axis` in case the `args` property doesn't work for some reason.